moveit_core/python/src/pymoveit.cpp

```diff
@@ -0,0 +1,55 @@
+#include <pybind11/pybind11.h>
+#include <urdf_parser/urdf_parser.h>
+#include <urdf_model/model.h>
+#include <random_numbers/random_numbers.h>
+#include <srdfdom/model.h>
+#include <moveit/python/pybind_rosmsg_typecasters.h>
+
+namespace py = pybind11;
+
+void def_collision_detect_bindings(py::module& contact);
+
+void def_robot_model_bindings(py::module& m);
+
+void def_robot_state_bindings(py::module& m);
+
+void def_transforms_bindings(py::module& m);
+
+void def_planning_scene_bindings(py::module& m);
+
+void def_kinematic_constraints_bindings(py::module& m);
+
+PYBIND11_MODULE(pymoveit, m)
+{
+  m.doc() = "moveit core";
+
+  auto collision_detection_m = m.def_submodule("collision_detection");
+  auto planning_scene_m = m.def_submodule("planning_scene");
+  auto robot_model_m = m.def_submodule("robot_model");
+  auto robot_state_m = m.def_submodule("robot_state");
+  auto transforms_m = m.def_submodule("transforms");
+  auto kinematic_constraints_m = m.def_submodule("kinematic_constraints");
+  auto urdf_m = m.def_submodule("urdf");
+  auto random_numbers_m = m.def_submodule("random_numbers");
+
+  py::class_<random_numbers::RandomNumberGenerator>(random_numbers_m, "RandomNumbers")
+      .def(py::init<>())
+      .def(py::init<boost::uint32_t>(), py::arg("seed"))
+      //
+      ;
+
+  def_collision_detect_bindings(collision_detection_m);
+  def_robot_model_bindings(robot_model_m);
+  def_robot_state_bindings(robot_state_m);
+  def_transforms_bindings(transforms_m);
+  def_planning_scene_bindings(planning_scene_m);
+  def_kinematic_constraints_bindings(kinematic_constraints_m);
+
+  urdf_m.def("parseURDFFile", &urdf::parseURDFFile);
+  py::class_<urdf::ModelInterface, urdf::ModelInterfaceSharedPtr>(urdf_m, "ModelInterface")
+      .def(py::init<>())
+      .def("getName", &urdf::ModelInterface::getName);
+
+  auto srdf_m = m.def_submodule("srdf", "srdf");
+  py::class_<srdf::Model>(srdf_m, "Model").def(py::init<>()).def("initFile", &srdf::Model::initFile);
```

Comment:
These definitions aren't yet useful, are they? Is there any need to expose these classes?

Reply:
True, as is they are useless. I'm expecting they will be exposed further as people need them. I don't see much benefit in deleting it, although separating into separate PRs could be reasonable if your concern is diff size.

Reply:
My main concern was that urdf, srdf, random_numbers are not parts of MoveIt, but have their own ROS packages.

Reply:
that's true. I think the initial impetus was to expose the constructor for RobotModel which takes a URDF and SRDF, and therefore we needed those classes exposed and usable, but then I decided to do that in C++ and write a little wrapper function that just takes the paths for the urdf and srdf filenames. Since the more common way will be to use the RobotModelLoader from the ros interface, which will eventually have python bindings, I don't think people will be constructing a RobotModel directly.

random_numbers is going to be needed by anyone who wants to control RNG when calling setToRandomPosition. Again maybe this isn't super common, but I felt like it was good to expose it anyways.

Of course the better approach would be to go write bindings in those packages, which we could certainly do if the maintainers there are interested.

Those are just my thoughts, of course it's up to you all, let me know if you'd like me to remove these all the not-actually-moveit bindings.

Reply:
I've removed all the not-actually-moveit bindings